### DISCUSSION TITLE
Multi-language resource management

### DISCUSSION THREAD

**Discussion by sedubois** [avo-hq/avo#2308] [Feature requests]
It appears Avo does not yet support creating resources in multiple languages or at least I did not find documentation for it (there is some [documentation](https://docs.avohq.io/3.0/localization.html) to localize the Avo UI using hard-coded translation strings, but that's not what is needed here). Avo describes itself as a "custom Content Management System" and I expect a CMS to support localization. For instance, Spina CMS seems to have implemented this using Mobility with JSONB backend and attr_json.

For starters, perhaps there could be just detailed documentation or a video tutorial to implement this using Avo from scratch.

It should be possible to translate entire resources, but ideally also only some specific fields. For instance, in a blog post, the main image might be the same for all languages but the title and body would differ. Each blog post, no matter the language, could also belong to several categories, but in turn the titles of those categories would themselves be localized. How to build this using Avo?

**Comment by adrianthedev** [comment]
Hey, @sedubois. Thanks for writing.

Avo doesn't yet have this feature as 1st party, but you can still achieve this in two ways.

1. If you have only a few languages, you can set up a language switcher using [this recipe](https://docs.avohq.io/3.0/recipes/multilingual-content.html)
2. If you have more than one language, you can use some of the setup from the recipe and render the language fields using something like this.

```ruby
[:en, :fr, :es, :de].each do |locale|
  field :"name_#{locale}", as: :string
end
```

This will render the fields for all locales.


We plan to support this use-case in the future, but it's not on our immediate roadmap.

**Reply by sedubois** [reply]
Thanks @adrianthedev for the documentation, I'll have a try 👍

**Comment by adrianthedev** [comment]
Closing for now. Feel free to re-open.
